packages/@expo/prebuild-config/src/plugins/icons/__tests__/withIosIcons-test.ts

```diff
@@ -177,3 +178,97 @@ describe('e2e: iOS icons with fallback image', () => {
     expect(contents.images.length).toBe(1);
   });
 });
+
+describe('iOS .icon directory', () => {
+  it('prefers ios-specific .icon over main icon', () => {
+    expect(
+      getIcons({
+        icon: './assets/main.png',
+        ios: {
+          icon: './assets/ios.icon',
+        },
+      })
+    ).toBe('./assets/ios.icon');
```

Comment:
we did use ios icon over the main icon previously, even the icon is not `.icon`. this new test case is not new though.